src/aro/Parser.zig

```diff
@@ -5574,7 +5574,11 @@ pub const Result = struct {
                 try res.implicitCast(p, .complex_float_to_complex_int);
             }
         } else if (!res.ty.eql(int_ty, p.comp, true)) {
-            try res.val.intCast(int_ty, p.comp);
+            const old_val = res.val;
+            const value_change_kind = try res.val.intCast(int_ty, p.comp);
+            if (value_change_kind == .value_changed and !(res.ty.makeIntegerUnsigned().specifier == int_ty.makeIntegerUnsigned().specifier))
```

Comment:
Is there a specific scenario the `!(res.ty.makeIntegerUnsigned().specifier == int_ty.makeIntegerUnsigned().specifier)` part guards against (probably better expressed using `!=` instead)?